### PR TITLE
Add font-width and alias font-stretch property

### PR DESCRIPTION
**Pull request by AlexsanderDamaceno**
#### 1b8e7931f7fa1174b80f6f65401b84139aa59bd0
<pre>
Add font-width and alias font-stretch property
<a href="https://bugs.webkit.org/show_bug.cgi?id=277673">https://bugs.webkit.org/show_bug.cgi?id=277673</a>

Reviewed by Antoine Quint.

Add font-width and alias font-stretch property

* Source/WebCore/css/FontFace.cpp:
(WebCore::FontFace::setWidth):
(WebCore::FontFace::setWeight): Deleted.
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::calculateWidthRange):
(WebCore::CSSFontFace::setWidth):
(WebCore::CSSFontFace::width const):
(WebCore::calculateStretchRange): Deleted.
(WebCore::CSSFontFace::setStretch): Deleted.
(WebCore::CSSFontFace::stretch const): Deleted.
* Source/WebCore/css/CSSFontFace.h:
* Source/WebCore/css/CSSFontFaceSet.cpp:
(WebCore::computeFontSelectionRequest):
(WebCore::CSSFontFaceSet::fontFace):
* Source/WebCore/css/CSSFontSelector.cpp:
(WebCore::CSSFontSelector::addFontFaceRule):
* Source/WebCore/css/CSSFontValue.cpp:
(WebCore::CSSFontValue::customCSSText const):
(WebCore::CSSFontValue::equals const):
(WebCore::CSSFontValue::customVisitChildren const):
* Source/WebCore/css/CSSFontValue.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForRepeatRule):
(WebCore::fontWidth):
(WebCore::fontShorthandValue):
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
(WebCore::fontStretch): Deleted.
* Source/WebCore/css/FontFace.cpp:
(WebCore::FontFace::create):
(WebCore::FontFace::setWeight):
(WebCore::FontFace::setWidth):
(WebCore::FontFace::width const):
(WebCore::FontFace::setStretch): Deleted.
(WebCore::FontFace::stretch const): Deleted.
* Source/WebCore/css/FontFace.h:
* Source/WebCore/css/FontFace.idl:
* Source/WebCore/css/FontSelectionValueInlines.h:
(WebCore::fontWidthKeyword):
(WebCore::fontWidthValue):
(WebCore::fontStretchKeyword): Deleted.
(WebCore::fontStretchValue): Deleted.
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::fontWidthKeyword):
(WebCore::ShorthandSerializer::serializeFont const):
(WebCore::fontStretchKeyword): Deleted.
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::CSSPropertyParser::consumeFont):
(WebCore::initialValueForLonghand):
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
(WebCore::CSSPropertyParserHelpers::parseFontFaceFontWidth):
(WebCore::CSSPropertyParserHelpers::consumeFontFaceFontWidth):
(WebCore::CSSPropertyParserHelpers::parseFontFaceFontStretch): Deleted.
(WebCore::CSSPropertyParserHelpers::consumeFontFaceFontStretch): Deleted.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
(WebCore::isValueOutOfRangeForProperty):
(WebCore::rangeForProperty):
* Source/WebCore/platform/graphics/FontCascadeDescription.h:
(WebCore::FontCascadeDescription::initialWidth):
(WebCore::FontCascadeDescription::initialStretch): Deleted.
* Source/WebCore/platform/graphics/FontDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.h:
(WebCore::FontDescription::width const):
(WebCore::FontDescription::setWidth):
(WebCore::FontDescription::stretch const): Deleted.
(WebCore::FontDescription::setStretch): Deleted.
* Source/WebCore/platform/graphics/FontSelectionAlgorithm.cpp:
(WebCore::FontSelectionAlgorithm::widthDistance const):
(WebCore::FontSelectionAlgorithm::indexOfBestCapabilities):
(WebCore::FontSelectionAlgorithm::stretchDistance const): Deleted.
* Source/WebCore/platform/graphics/FontSelectionAlgorithm.h:
(WebCore::ultraCondensedWidthValue):
(WebCore::extraCondensedWidthValue):
(WebCore::condensedWidthValue):
(WebCore::semiCondensedWidthValue):
(WebCore::normalWidthValue):
(WebCore::semiExpandedWidthValue):
(WebCore::expandedWidthValue):
(WebCore::extraExpandedWidthValue):
(WebCore::ultraExpandedWidthValue):
(WebCore::FontSelectionSpecifiedCapabilities::computeWidth const):
(WebCore::ultraCondensedStretchValue): Deleted.
(WebCore::extraCondensedStretchValue): Deleted.
(WebCore::condensedStretchValue): Deleted.
(WebCore::semiCondensedStretchValue): Deleted.
(WebCore::normalStretchValue): Deleted.
(WebCore::semiExpandedStretchValue): Deleted.
(WebCore::expandedStretchValue): Deleted.
(WebCore::extraExpandedStretchValue): Deleted.
(WebCore::ultraExpandedStretchValue): Deleted.
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::capabilitiesForFontDescriptor):
* Source/WebCore/platform/graphics/skia/FontCacheSkia.cpp:
(WebCore::skiaFontStyle):
* Source/WebCore/platform/graphics/skia/FontCustomPlatformDataSkia.cpp:
(WebCore::FontCustomPlatformData::fontPlatformData):
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
(WebCore::traitsInFamilyEnumProc):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
(WebCore::RenderStyle::setFontWidth):
(WebCore::RenderStyle::setFontStretch): Deleted.
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::fontWidth const):
(WebCore::RenderStyle::fontStretch const): Deleted.
* Source/WebCore/style/PropertyAllowlist.cpp:
(WebCore::Style::isValidMarkerStyleProperty):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertFontWidthFromValue):
(WebCore::Style::BuilderConverter::convertFontWidth):
(WebCore::Style::BuilderConverter::convertFontStretchFromValue): Deleted.
(WebCore::Style::BuilderConverter::convertFontStretch): Deleted.
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::cssPropertyIdForSVGAttributeName):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertFontWidth):
(WebCore::Style::BuilderConverter::convertFontStretch): Deleted.
* Source/WebCore/style/StyleResolveForFont.cpp:
(WebCore::Style::fontStretchFromCSSValueDeprecated):
(WebCore::Style::fontStretchFromCSSValue):
* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::systemFontParameters):
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
(WebCore::CSSPropertyParserHelpers::consumeFontFaceFontWidth):
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:

* LayoutTests/fast/css/font-shorthand-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-font-family-expected.txt:
* LayoutTests/fast/text/font-face-javascript-expected.txt:
* LayoutTests/fast/text/font-loading-global-keyword-expected.txt:
* LayoutTests/fast/text/font-selection-font-face-parse-expected.txt:
* LayoutTests/fast/text/variations/font-loading-api-parse-ranges-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-inner-at-rules-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/animations/font-stretch-interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/font-face-range-order-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/at-font-face-descriptors-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/font-shorthand-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* Source/WebCore/css/FontFace.idl:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::cssPropertyIdForSVGAttributeName):
* Source/WebCore/svg/svgattrs.in:
* Source/WebCore/animation/KeyframeEffect.cpp:
(WebCore::IDLAttributeNameToAnimationPropertyName):

Canonical link: <a href="https://commits.webkit.org/288000@main">https://commits.webkit.org/288000@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8dbfc0615c24df9c823bb358c928199b27a3d573

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81620 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1144 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35568 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86153 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32610 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/83726 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1172 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/8961 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63680 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21406 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84690 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/826 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74270 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43969 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/726 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28449 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31065 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72154 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29044 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87596 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/8857 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6274 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72002 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9039 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70097 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71237 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17740 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15317 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14231 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8809 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14337 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8649 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12170 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10456 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/css/CSSProperties.json [hunk 1]
```diff
@@ -647,6 +647,7 @@
                     "parser-grammar": "<font-stretch-absolute> | <percentage [0,inf]>"
                 },
                 {
+                    "aliases": ["font-witdh"],
```
Comment (nt1m) [inline Source/WebCore/css/CSSProperties.json]:
```suggestion
                    "aliases": [
                        "font-stretch"
                    ],
```

File: Source/WebCore/css/CSSProperties.json [hunk 2]
```diff
@@ -9626,7 +9627,37 @@
                     "url": "https://www.w3.org/TR/css-fonts-4/#descdef-font-face-font-feature-settings"
                 }
             },
-            "font-stretch": {
+          
+            "font-style": {
```
Comment (nt1m) [inline Source/WebCore/css/CSSProperties.json]: This addition seems unrelated to this change

File: Source/WebCore/css/ComputedStyleExtractor.cpp [hunk 3]
```diff
@@ -3646,7 +3646,7 @@ RefPtr<CSSValue> ComputedStyleExtractor::valueForPropertyInStyle(const RenderSty
         return fontSizeAdjustFromStyle(style);
     case CSSPropertyFontStyle:
         return fontStyle(style);
-    case CSSPropertyFontStretch:
+    case CSSPropertyFontWidth:
         return fontStretch(style);
```
Comment (nt1m) [inline Source/WebCore/css/ComputedStyleExtractor.cpp]: rename this method as well?

File: Source/WebCore/css/CSSFontSelector.cpp [hunk 4]
```diff
@@ -180,7 +180,7 @@ void CSSFontSelector::addFontFaceRule(StyleRuleFontFace& fontFaceRule, bool isIn
     RefPtr familyList = dynamicDowncast<CSSValueList>(style.getPropertyCSSValue(CSSPropertyFontFamily));
     RefPtr fontStyle = style.getPropertyCSSValue(CSSPropertyFontStyle);
     RefPtr fontWeight = style.getPropertyCSSValue(CSSPropertyFontWeight);
-    RefPtr fontStretch = style.getPropertyCSSValue(CSSPropertyFontStretch);
+    RefPtr fontStretch = style.getPropertyCSSValue(CSSPropertyFontWidth);
```
Comment (nt1m) [inline Source/WebCore/css/CSSFontSelector.cpp]: rename this variable?

File: Source/WebCore/animation/CSSPropertyAnimation.cpp [hunk 5]
```diff
@@ -3892,7 +3892,7 @@ CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap()
 #endif
         new PropertyWrapperFontSizeAdjust,
         new PropertyWrapperFontWeight,
-        new PropertyWrapper<FontSelectionValue>(CSSPropertyFontStretch, &RenderStyle::fontStretch, &RenderStyle::setFontStretch),
+        new PropertyWrapper<FontSelectionValue>(CSSPropertyFontWidth, &RenderStyle::fontStretch, &RenderStyle::setFontStretch),
```
Comment (nt1m) [inline Source/WebCore/animation/CSSPropertyAnimation.cpp]: You should rename the setters/getters on render style too

File: Source/WebCore/css/ShorthandSerializer.cpp [hunk 6]
```diff
@@ -931,7 +931,7 @@ String ShorthandSerializer::serializeFont() const
     auto styleIndex = longhandIndex(0, CSSPropertyFontStyle);
     auto capsIndex = longhandIndex(1, CSSPropertyFontVariantCaps);
     auto weightIndex = longhandIndex(2, CSSPropertyFontWeight);
-    auto stretchIndex = longhandIndex(3, CSSPropertyFontStretch);
+    auto stretchIndex = longhandIndex(3, CSSPropertyFontWidth);
```
Comment (nt1m) [inline Source/WebCore/css/ShorthandSerializer.cpp]: Can you rename this variable?

File: Source/WebCore/css/CSSFontFace.cpp [hunk 7]
```diff
@@ -376,7 +376,7 @@ String CSSFontFace::weight() const
 
 String CSSFontFace::stretch() const
```
Comment (nt1m) [inline Source/WebCore/css/CSSFontFace.cpp]: This should probably be renamed

File: Source/WebCore/css/CSSFontFace.cpp [hunk 8]
```diff
@@ -220,7 +220,7 @@ static FontSelectionRange calculateStretchRange(CSSValue& value)
 
 void CSSFontFace::setStretch(CSSValue& style)
 {
-    mutableProperties().setProperty(CSSPropertyFontStretch, &style);
+    mutableProperties().setProperty(CSSPropertyFontWidth, &style);
 
     auto range = calculateStretchRange(style);
```
Comment (nt1m) [inline Source/WebCore/css/CSSFontFace.cpp]: These should probably renamed

**Review by nt1m — COMMENTED** [review]
There's probably more to alias / rename

File: Source/WebCore/css/FontFace.idl [hunk 9]
```diff
@@ -53,7 +53,7 @@ dictionary FontFaceDescriptors {
     [SetterCallWith=CurrentScriptExecutionContext] attribute DOMString family;
     [SetterCallWith=CurrentScriptExecutionContext] attribute DOMString style;
     [SetterCallWith=CurrentScriptExecutionContext] attribute DOMString weight;
-    [SetterCallWith=CurrentScriptExecutionContext] attribute DOMString stretch;
+    [SetterCallWith=CurrentScriptExecutionContext] attribute DOMString width;
```
Comment (nt1m) [inline Source/WebCore/css/FontFace.idl:56]:
```suggestion
    [SetterCallWith=CurrentScriptExecutionContext] attribute DOMString width;
    [ImplementedAs=width, SetterCallWith=CurrentScriptExecutionContext] attribute DOMString stretch;
```

We also need to alias stretch here too.

File: Source/WebCore/svg/SVGElement.cpp [hunk 10]
```diff
@@ -900,7 +900,7 @@ CSSPropertyID SVGElement::cssPropertyIdForSVGAttributeName(const QualifiedName&
     case AttributeNames::font_size_adjustAttr:
         return CSSPropertyFontSizeAdjust;
     case AttributeNames::font_stretchAttr:
-        return CSSPropertyFontStretch;
+        return CSSPropertyFontWidth;
```
Comment (nt1m) [inline Source/WebCore/svg/SVGElement.cpp]: Can you add font-width here: https://searchfox.org/wubkat/rev/b36cbce69fddb7da33823f316bd8ead5bebee970/Source/WebCore/svg/svgattrs.in#62 ?

and also add a mapping for font-width here ?

**Review by nt1m — COMMENTED** [review]
Looks fine for most part, thanks for doing this! Once the 2 suggested changes are made and locally built, can you rebaseline the tests accordingly (e.g. run the failing tests with `./Tools/Scripts/run-webkit-tests path/to/test --reset-results`)

**Comment by nt1m** [comment]
@AlexsanderDamaceno Any update on this? This looks close to the finition, aside from my 2 comments

File: Source/WebCore/css/CSSProperties.json [hunk 11]
```diff
@@ -620,7 +620,7 @@
                 "url": "https://www.w3.org/TR/css-fonts-4/#font-weight-prop"
             }
         },
-        "font-stretch": {
+        "font-width": {
```
Comment (nt1m) [inline Source/WebCore/css/CSSProperties.json:682]: The @font-face at-rule needs a similar change: https://searchfox.org/wubkat/rev/e17ffe7b5bd3105bdd72617c07655ee4f2d0fdd4/Source/WebCore/css/CSSProperties.json#9679

(along with `consumeFontFaceFontStretch`)

File: Source/WebCore/css/FontFace.idl [hunk 12]
```diff
@@ -37,7 +37,7 @@ enum FontFaceLoadStatus {
 dictionary FontFaceDescriptors {
     DOMString style = "normal";
     DOMString weight = "normal";
-    DOMString stretch = "normal";
+    DOMString width = "normal";
```
Comment (nt1m) [inline Source/WebCore/css/FontFace.idl]: This needs a similar change to below in order to fix the remaining JS API tests.

```
    DOMString width = "normal";
    [ImplementedAs=width] DOMString stretch;
```

I think this might work ?

**Review by nt1m — COMMENTED** [review]
I left comments about what I think the remaining failures are about.

File: Source/WebCore/css/FontFace.idl (hunk 12 above)
Comment (AlexsanderDamaceno) [reply Source/WebCore/css/FontFace.idl]: @nt1m the font-face sugestion make some test pass, but still problem with the this one, is not mapping yet.

Comment (AlexsanderDamaceno) [reply Source/WebCore/css/FontFace.idl]: Solved the problem making` [ImplementedAs=width] DOMString stretch = "normal"` and removing the  `DOMString width = "normal";`

Comment (nt1m) [reply Source/WebCore/css/FontFace.idl]: The `width` property also needs to work in the font-face constructor

**Comment by nt1m** [comment]
@graouts I'd be curious if you know why `font-stretch` (now an alias of `font-width`) does not animate properly with the Web Animations API: https://ews-build.webkit.org/#/builders/63/builds/20483

Is the Web Animations API broken when you use aliases?

File: Source/WebCore/css/FontFace.idl (hunk 12 above)
Comment (nt1m) [reply Source/WebCore/css/FontFace.idl]: Perhaps the order matters:

```
    [ImplementedAs=width] DOMString stretch;
    DOMString width = "normal";
```

**Comment by nt1m** [comment]
This is the codepath Web Animations use: https://searchfox.org/wubkat/rev/d45649fa32a499702c30474e94bd6843c4c67849/Source/WebCore/dom/Element.cpp#5674

Presumably there is some missing handling of aliases happening somewhere.

**Comment by graouts** [comment]
> @graouts hey, could you give a look at this, we are working to alias font-stretch with font-width, almost all cases passing, but one is showing problem with web animation, do you have some idea about what could be wrong ?

Which test is it? Is it using the Web Animations JavaScript API? If so, you can debug from under `KeyframeEffect::processKeyframes()` and see if we're picking up aliased CSS properties correctly.

**Comment by nt1m** [comment]
> > @graouts hey, could you give a look at this, we are working to alias font-stretch with font-width, almost all cases passing, but one is showing problem with web animation, do you have some idea about what could be wrong ?
> 
> Which test is it? Is it using the Web Animations JavaScript API? If so, you can debug from under `KeyframeEffect::processKeyframes()` and see if we're picking up aliased CSS properties correctly.

The test is `imported/w3c/web-platform-tests/css/css-fonts/animations/font-stretch-interpolation.html`, we're not picking up on interpolating `font-stretch` now that it's defined as an alias of `font-width`

**Comment by graouts** [comment]
@AlexsanderDamaceno I don't think we can remove that check. Instead we should find a way to rework it such that it knows about aliasing, or more specifically about this property, just like we have custom code to deal with `offset` and `float`.

File: Source/WebCore/animation/KeyframeEffect.cpp [hunk 13]
```diff
@@ -128,8 +128,11 @@ static inline CSSPropertyID IDLAttributeNameToAnimationPropertyName(const AtomSt
     // 3. If attribute is the string "cssOffset", then return an animation property representing the CSS offset property.
     if (idlAttributeName == "cssOffset"_s)
         return CSSPropertyOffset;
+    // 4. If the attribute is the string "fontStretch" return the CSS offset property that it aliases.
```
Comment (graouts) [inline Source/WebCore/animation/KeyframeEffect.cpp]: The numbering comes from the Web Animations spec, so you can't increase it here. Also, you copied and pasted "offset" here.
```suggestion
    // If the attribute is the string "fontStretch" return the "font-width" property that it aliases.
```

File: LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt [hunk 14]
```diff
@@ -278,8 +278,6 @@ PASS flood-color supports animating as color of hsla()
 PASS flood-opacity (type: opacity) has testAccumulation function
 PASS flood-opacity: [0, 1] number
 PASS flood-opacity: [0, 1] number (clamped)
-PASS font-stretch (type: percentage) has testAccumulation function
-PASS font-stretch: percentage
```
Comment (graouts) [inline LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt]: So with the aliasing, I suppose this should be kept, but we should be adding an entry for `font-width` _as well_. The file in question is `LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js`. If you do make that change, you'll need to make sure it's exported to the WPT repository. You can do that manually or using the `Tools/Scripts/export-w3c-test-changes` script.

Comment (nt1m) [reply LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt]: I agree

Comment (AlexsanderDamaceno) [reply LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt]: I included font-width in property-list.js and remove the mistake of numbers in `IDLAttributeNameToAnimationPropertyName.`
Once you review property-list.js and confirm it's correct, I will export it to the WPT repository.

**Comment by nt1m** [comment]
Looks fine! Can you update the baselines that EWS shows as failing?

File: LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js [hunk 15]
```diff
@@ -564,6 +564,10 @@ const gCSSProperties1 = {
     // https://drafts.csswg.org/css-fonts-4/#propdef-font-stretch
     types: [ 'percentage' ]
   },
+  'font-width': {
+    // https://drafts.csswg.org/css-fonts-4/#propdef-font-width
+    types: [ 'percentage' ]
+  },
```
Comment (graouts) [inline LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js]: I _think_ that file sorts its properties alphabetically, so it might be better to move this further down.

**Review by graouts — APPROVED** [review]
